src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -163,7 +175,7 @@ public String getAttributeValue(String namespace, String localPart) {
      */
     @Override
     public String getLocalPart() {
-        return text;
+        return TokenUtil.getTokenName(detailAst.getType());
```

Comment:
can we cache `text` value ? we can do `getTokenName` ones.

Reply:
No. This code
```java
if (text == null) {
  text = TokenUtil.getTokenName(detailAst.getType());
}
return text;
```
does not pass the Pitest regression. It says "hey, I replaced `if (text == null)` with `false` and nothing fails.
This is common to all nano-optimizations. The method `TokenUtil.getTokenName` is O(1), so there is no effect for such cache.

Reply:
there is nothing quicker then just return reference to String.
but ok, this optimization is ok to skip for now.